Web UI env var reference (#1640)
temporalio/documentation: docs/references/docker-env-vars.md

---
id: docker-env-vars
title: Web UI Docker Environmental Variables
sidebar_label: Web UI Docker Environmental Variables
description: How to set the docker environmental variables quickly.
tags:
  - docker
  - webui
---

Docker containers can be configured for use in a production setting.
Use `docker run` to configure the Web UI environmental variables.

```
docker run \
    -e TEMPORAL_ADDRESS=127.0.0.1:7233 \
    -e TEMPORAL_UI_PORT=8080 \
    -e TEMPORAL_AUTH_ENABLED=true \
    -e TEMPORAL_AUTH_PROVIDER_URL=https://accounts.google.com \
    -e TEMPORAL_AUTH_CLIENT_ID=xxxxx-xxxx.apps.googleusercontent.com \
    -e TEMPORAL_AUTH_CLIENT_SECRET=xxxxxxxxxxxxxxx \
    -e TEMPORAL_AUTH_CALLBACK_URL=https://xxxx.com:8080/auth/sso/callback \
    -e TEMPORAL_UI_ENABLED=true \
    -e TEMPORAL_OPENAPI_ENABLED=true \
    -e TEMPORAL_TLS_CA=../ca.cert \
    -e TEMPORAL_TLS_CERT=../cluster.pem \
    -e TEMPORAL_TLS_KEY=../cluster.key \
    -e TEMPORAL_TLS_ENABLE_HOST_VERIFICATION=true \
    -e TEMPORAL_TLS_SERVER_NAME=tls-server \
    temporalio/ui:<tag>
```

The environmental variables are defined below:

> - [`TEMPORAL_ADDRESS`](#temporal-address) : Temporal frontend address.

Reviewer: Why are these linked ? Where are they linking to?

- `TEMPORAL_UI_PORT` : port to serve HTTP API and UI
- `TEMPORAL_AUTH_ENABLED` : enable or disable auth
- `TEMPORAL_AUTH_PROVIDER_URL` : Auth OIDC provider URL
- `TEMPORAL_AUTH_CLIENT_ID` : auth client ID
- `TEMPORAL_AUTH_CLIENT_SECRET` : auth client secret
- `TEMPORAL_AUTH_CALLBACK_URL` : auth callback URL
- `TEMPORAL_UI_ENABLED` : enable or disable serve UI
- `TEMPORAL_OPENAPI_ENABLED` : enable or disable serve open API UI
- `TEMPORAL_TLS_CA` : TLS Certificate Authority path
- `TEMPORAL_TLS_CERT` : TLS cert path

> - `TEMPORAL_TLS_KEY` : TLS key path

Author: they are paths.

---
id: docker-env-vars

> title: Web UI Docker Environmental Variables
> sidebar_label: Web UI Docker Environmental Variables
> description: How to set the docker environmental variables quickly.

Reviewer 2: ```suggestion
title: Docker environmental variables for Web UI
sidebar_label: Docker environmental variables
description: How to set Docker environmental variables for Temporal Web UI.
```

tags:
  - docker
  - webui
---

Docker containers can be configured for use in a production setting.
Use `docker run` to configure the Web UI environmental variables.

```
docker run \
    -e TEMPORAL_ADDRESS=127.0.0.1:7233 \
    -e TEMPORAL_UI_PORT=8080 \
    -e TEMPORAL_AUTH_ENABLED=true \
    -e TEMPORAL_AUTH_PROVIDER_URL=https://accounts.google.com \
    -e TEMPORAL_AUTH_CLIENT_ID=xxxxx-xxxx.apps.googleusercontent.com \
    -e TEMPORAL_AUTH_CLIENT_SECRET=xxxxxxxxxxxxxxx \
    -e TEMPORAL_AUTH_CALLBACK_URL=https://xxxx.com:8080/auth/sso/callback \
    -e TEMPORAL_UI_ENABLED=true \
    -e TEMPORAL_OPENAPI_ENABLED=true \
    -e TEMPORAL_TLS_CA=../ca.cert \
    -e TEMPORAL_TLS_CERT=../cluster.pem \
    -e TEMPORAL_TLS_KEY=../cluster.key \
    -e TEMPORAL_TLS_ENABLE_HOST_VERIFICATION=true \
    -e TEMPORAL_TLS_SERVER_NAME=tls-server \
    temporalio/ui:<tag>
```

> ## Environmental Variables

Reviewer: Nix this -

The environmental variables are defined as follows:

> - `TEMPORAL_ADDRESS` : Temporal frontend address.

Reviewer: ```suggestion
## `TEMPORAL_ADDRESS`

Temporal frontend address.
```

Do this ^

---

> id: docker-env-vars

Reviewer: The id and the file name should probably be `web-ui-server-env-vars` - as that focuses on the web ui - and generalizes the env vars (because they could be docker or container or other envs)
